js/pages/settings.js

```diff
@@ -477,6 +527,123 @@ const refreshUserDataAfterEdit = async () => {
   }
 };
 
+/**
+ * Sign-in Information -> removeLoginEmailButton and removeLoginPhoneButton.
+ * Show the confirmation modal after user clicks 'Remove this <ButtonType>' button.
+ * Require confirmation from user prior to unlinking the authentication provider.
+ * Then close the modal.
+ */
+const attachLoginEditFormButtons = async (currentEmail, currentPhone) => {
+    let removalType = null;
+
+    const modalMap = {
+        'Email': currentEmail ? document.getElementById('confirmationModalEmail') : null,
+        'Phone': currentPhone ? document.getElementById('confirmationModalPhone') : null
+    }
+
+    const modalStatusMap = {
+        'Email': modalMap['Email'] && modalMap['Email'].style.display === 'block',
+        'Phone': modalMap['Phone'] && modalMap['Phone'].style.display === 'block'
+    }
+
+    const openModal = (type) => {
+        if (!modalStatusMap[type]) {
+            modalMap[type].style.display = 'block';
+            removalType = type;
+            modalStatusMap[type] = true;
+        }
+    }
+
+    const closeModal = (type) => {
+        modalMap[type].style.display = 'none';
+        modalStatusMap[type] = false;
+    }
+
+    /**
+     * Add event listeners to update login form buttons.
+     * Handle modal toggling based on the active modal (phone or email)
+     * Authentication method protection: Only trigger unlink operation if the user already has another login method in firebase auth.
+     * If the user has only one login method, then do not allow the user to unlink it, show alert in this case.
+     * @param {string} type - 'Email' or 'Phone' 
+     * @param {string} buttonID - the HTML Button's id: 'removeLoginEmailButton' or 'removeLoginPhoneButton' 
+     * @param {string} confirmButtonID - the HTML Button's id: 'confirmRemoveEmailButton' or 'confirmRemovePhoneButton'
+     * @param {string} cancelRemoveButtonID - the HTML Button's id: 'cancelRemoveEmailButton' or 'cancelRemovePhoneButton'
+     */
+    const addListenerToButton = async (type, buttonID, confirmButtonID, cancelRemoveButtonID) => {
```

Comment:
My comment is for discussing the approach of having nested function declarations inside a main function. The main function declaration is `attachLoginEditFormButtons`. The nested function declarations are the following: 
- `openModal`
- `closeModal`
- `addListenerToButton`
- `updateUIAfterUnlink`

Normally function declarations are placed outside and a function call is made in one place or multiple places. In this specific scenario nested functions are declared and called inside the main function, `attachLoginEditFormButtons`. To my understanding encapsulation is done here so the nested functions are not accessible outside of the main function. Also, is this approach done to take advantage of [lexical scope](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures)? I see the `modalMap` variable belonging to the parent function, `attachLoginEditFormButtons`, is referenced on [line 573](https://github.com/episphere/connectApp/pull/564/files#diff-7305c0ddbacbdf6b9110d08cd1e5de4b6bb477c81f635e5e700938a6c3a2eef6R573) and used to determine if the conditional block runs. I assume by accessing the parent scope, the nested function declarations can have their parameter size reduced as well as the function call arguments.

Reply:
Prior to reading the code, I may have moved the nested functions outside. I agree with the idea of bundling the logic in one function and the approach taken. Thank you for the explanation.